cli/command/stack/list.go

```diff
@@ -35,7 +35,8 @@ func newListCommand(dockerCli command.Cli, common *commonOptions) *cobra.Command
 	return cmd
 }
 
-func runList(cmd *cobra.Command, dockerCli command.Cli, opts options.List, orchestrator command.Orchestrator) error {
+// RunList performs a stack list against the specified orchestrator
+func RunList(cmd *cobra.Command, dockerCli command.Cli, opts options.List, orchestrator command.Orchestrator) error {
```

Comment:
Only concern I have with exporting all of these is that;

- These functions now become a public "API", with no interface defined for them
  - this can complicate making changes in the CLI (they were kept non-exported, as their signature/interface was not "stable")
- Some of these functions actually have a "UX" component built-in (e.g. they print confirmation messages on stdout/stderr, have an option to print formatted output); is that really suitable for external consumption?
  - calling these commands would become almost equivalent to `exec.Cmd("docker stack ls")`


For the _underlying_ code on the other hand, we have various interfaces defined; if those are lacking functionality that's needed, should we review those interfaces, and make that the API to use?

Reply:
was discussing with @chris-crone to add a note somewhere (perhaps the readme) to indicate that these functions, even though they're exported, should not be considered a stable API (as in; their signature can change, and shouldn't be relied on)

We could also add notes to the GoDoc of these, similar to https://golang.org/pkg/testing/#RunTests

> An internal function but exported because it is cross-package; part of the implementation of the "go test" command.

But that would be a bit more work to do

Reply:
> was discussing with @chris-crone to add a note somewhere (perhaps the readme) to indicate that these functions, even though they're exported, should not be considered a stable API (as in; their signature can change, and shouldn't be relied on)

Will do that in a follow-up 👍